p4include/dpdk/psa.p4

```diff
@@ -0,0 +1,863 @@
+/* Copyright 2013-present Barefoot Networks, Inc.
```

Comment:
Is this psa.p4 different from the other one?
If not, does it have to be here?

Reply:
Yes. This is slightly different from the standard psa.p4 in p4include directory.

This psa.p4 extends the API for Meter execute and Counter count methods. A new optional parameter is added to both of these methods to take packet length/counter increment value  as a parameter.

Meter extern's execute method:
`PSA_MeterColor_t execute(in S index, in PSA_MeterColor_t color, @optional in bit<32> pkt_len);`

Counter extern's count method
`void count(in S index, @optional in bit<32> increment);`

This API is required to meet DPDK target's requirements.

Reply:
Please also refer to https://github.com/p4lang/p4c/pull/2751 and https://github.com/p4lang/p4c/pull/2818 PRs which proposes similar extensions to PSA

Reply:
I find this very confusing. The whole point of PSA is to be portable. If we have two different PSA.p4 I don't see how portability is helped. Only confusion can ensue. 
We should find a resolution to the issues you cite before merging this file.
I would expect that extensions can be done by including psa.p4 and then additional libraries. But the externs in these libraries would have to have different names.
In the meantime we could perhaps use a different name for this file.

Reply:
What is the relation of this file and the one from https://github.com/p4lang/p4c/pull/2818?